CHANGELOG.md

```diff
@@ -62,6 +62,8 @@ to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 * Timestamp objects in templates now have `after(date) -> Boolean` and
   `before(date) -> Boolean` methods for comparing timestamps to other dates.
 
+* `jj help` now can give help for some categories, like `jj help revsets`.
```

Comment:
I don't think that the categories should be at the first level of help, especially since some of the names will be the same as revset functions or template functions or even aliases.
If you look at Git, some commands are listed and at the bottom is
```
'git help -a' and 'git help -g' list available subcommands and some
concept guides. See 'git help <command>' or 'git help <concept>'
to read about a specific subcommand or concept.
See 'git help git' for an overview of the system.
```
If you look at Breezy, a few commands are listed and at the bottom is
```
  brz help init      more help on e.g. init command
  brz help commands  list all commands
  brz help topics    list all help topics
```

Reply:
I don't have a strong opinion, but it seems also good to mention `jj help --help` or `jj help --list-categories` in the top-level help. Then, the categories list can be removed from the top-level help.

Reply:
I don't have a strong opinion either, but we should probably decide before we add support for topics and add a top-level `jj help topics` about that. We should decide if we want `jj help topics` to be about the "topics" feature or if we want `jj help topics topics` to be for that.

Reply:
I feel `jj help topics <TOPIC>` is too verbose. I prefer `jj help <--command|--topic> NAME..` (like `man [PAGE] NAME`) or `jj help <--topic=NAME|COMMAND..>` instead.

Reply:
I prefer the `--topic` flag too. Just to be clear, we prefer `topic` over `category`?

Reply:
I think “topic” is going to be really confusing once topics are a thing.

Reply:
Are topics different from categories? I'm a bit confused here XD

Reply:
We're going to introduce "topic" branches concept.

`--category` sounds also good, but it's short form would conflict with `-c|--command` if we add it. Mercurial uses `-k|--keyword`.

Reply:
Ohhh, I see. Where can learn more about this feature?

I'm fine changing to `keyword` to have the flag.

Reply:
Git uses `--g` for `guides`.

Reply:
> Where can learn more about this feature?
https://github.com/martinvonz/jj/issues/3402

Reply:
I think `keyword` is more generic.

Reply:
I made big changes, `s/categories/keywords/` and now we have a boolean flag '--keyword' to select between help for commands or keywords.